src/multiple-choice/components/app.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useLARAInteractiveAPI } from "../../shared/hooks/use-lara-interactive-api";
+import { Authoring } from "./authoring";
+import { Runtime } from "./runtime";
+
+export const App = () => {
+  const [ aspectRatio, setAspectRatio ] = useState(1);
+  const interactiveAPIConfig = {
+    interactiveState: true,
+    authoredState: true,
+    aspectRatio
+  };
+  const { mode, authoredState, interactiveState, setInteractiveState, setAuthoredState } = useLARAInteractiveAPI(interactiveAPIConfig);
+  const container = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    // Measure the main app container on every re-render and update aspect ratio. Note that useLARAInteractiveAPI
+    // takes care of its updates and will notify LARA about it using iframe phone.
+    const width = container.current?.offsetWidth;
+    const height = container.current?.offsetHeight
+    if (width && width > 0 && height && height > 0) {
+      setAspectRatio(width / height);
+    }
+  });
```

Comment:
I guess since this useEffect doesn't have a list of dependencies it will run on every render. Will the `useRef` trigger a re-render if the interactive is embedded in a responsive layout page and the page is resized?

Reply:
Yes, this will run on every render. So, if some component is updated, user types, clicks, and so on. Usually, that's actions that cause size change (e.g. when the user adds a new choice in the form). It's a little heuristic, but I think that it should be working in most cases (a case where it won't work - user trying to resize textarea, as that doesn't trigger state change and rendering).

Re iframe resizing in the responsive layout - no, useRef doesn't monitor anything and doesn't trigger any renders. 

I thought this question would be used in the assessment block usually. But I think that interactive should always report its own aspect ratio and it shouldn't really care about size of the page or iframe. In this case - the question content aspect ratio. So, I don't think that page/iframe size should matter here. Even if the iframe is too small to display interactive completely and there're scroll bars, the aspect ratio will be reported correctly - I tested that and that's how it works while we're authoring and adding new choices. I guess for some milliseconds interactive is larger than LARA iframe, but interactive immediately sends new aspect ratio, LARA accepts that, and in the end we can see that height is increased and there're no scrollbars.

We could simply add document/window.resize handler here. But I think that page/window resize won't change the aspect ratio that we calculate (or at least shouldn't).

Reply:
Yeah it is an edge case that this would be added to the interactive box and thus affected by responsive layout. However we should handle it because it is possible.
If you switch to using height api instead of aspect ratio this will also simplify things.
But I suspect in either case the height or aspect ratio will want to change when the window is resized. For example a large prompt that spans multiple lines will reflow when the page is made more narrow or wide. So that should trigger a height or aspect ratio change.

Reply:
I made a test here to see what happens: https://authoring.staging.concord.org/activities/20598/pages/307321
It is interesting how the interactive starts to get shorter instead of longer as the page width is shrunk.  I guess that makes sense because the aspect ratio doesn't change so with less width there is less height too.